include/seqan3/alignment/configuration/global_config.hpp

```diff
@@ -0,0 +1,121 @@
+
+namespace seqan3::detail
+{
+
+// ----------------------------------------------------------------------------
+// global_alignment_cfg
+// ----------------------------------------------------------------------------
+
+//!\brief The global_alignment_configuration element to distinguish between alignment policies.
+class global_alignment_cfg : public config_element_base<global_alignment_cfg>
+{
+    //!\brief Grant access to private state.
+    friend class config_element_access<global_alignment_cfg>;
+
+    //!\brief State is ignored.
+    remove_cvref_t<decltype(std::ignore)> state;
+};
+}  // namespace seqan3::detail
+
+namespace seqan3::align
+{
+
+/*!\brief Global alignment configuration object.
+ * \ingroup align
+ */
+template <seqan3::detail::config_element_concept ... cfg_elements_t>
+class global_config : public detail::configuration<cfg_elements_t...>
+{
+    //!\brief The base type.
+    using base_type = detail::configuration<cfg_elements_t...>;
+
+public:
+    /*!\name Constructor, destructor and assignment
+     * \{
+     */
+    global_config()                                  = delete;
+    global_config(global_config const &)             = default;
+    global_config(global_config &&)                  = default;
+    global_config & operator=(global_config const &) = default;
+    global_config & operator=(global_config &&)      = default;
+    ~global_config()                                 = default;
+
+    //TODO(rrahn): Using the intermediate concept form here causes an ICE!
+    //!\brief Construction from another seqan3::detail::configuration object.
+    template <typename ... other_configs_t>
+    //!\cond
+        requires (detail::config_element_concept<other_configs_t> && ...)
+    //!\endcond
+    global_config(detail::configuration<other_configs_t...> const & other) : base_type{other}
+    {}
+
+    //TODO(rrahn): Using the intermediate concept form here causes an ICE!
+    //!\brief Construction from another seqan3::detail::configuration object.
+    template <typename ... other_configs_t>
+    //!\cond
+        requires (detail::config_element_concept<other_configs_t> && ...)
+    //!\endcond
+    global_config(detail::configuration<other_configs_t...> && other) : base_type{std::move(other)}
+    {}
+    //!}
+};
+
+/*!\name Deduction guides
+ * \relates seqan3::align::global_config
+ * \{
+ */
+
+//!\brief Deduces template arguments from constructing with another seqan3::detail::configuration.
+template <seqan3::detail::config_element_concept ... cfg_elements_t>
+global_config(detail::configuration<cfg_elements_t...> const &) ->
+    global_config<detail::global_alignment_cfg, cfg_elements_t...>;
+
+//!\brief Deduces template arguments from constructing with another seqan3::detail::configuration.
+template <seqan3::detail::config_element_concept ... cfg_elements_t>
+global_config(detail::configuration<cfg_elements_t...> &&) ->
+    global_config<detail::global_alignment_cfg, cfg_elements_t...>;
+//!\}
```

Comment:
Shouldn't the previous one be sufficient?

Reply:
If I think about it, one could also have the one ctor taking by copy and always move. Requires all config elements to be movable. I think we can assume that. As long as std::reference_wrapper is movable. Have to check

Reply:
But these are not constructors, they are deduction guides, and since both deduce to the same type, one should be sufficient (independent of anything else).